lonboard/experimental/_layer.py

```diff
@@ -329,3 +357,216 @@ class TextLayer(BaseArrowLayer):
+    """Whether or not the path fades out.
+
+    - Type: `bool`, optional
+    - Default: `True`
+    """
+
+    trail_length = traitlets.Float(default_value=None, allow_none=True).tag(sync=True)
+    """Trail length.
+
+    - Type: `float`, optional
+    - Default: `120`
+    """
+
+    current_time = traitlets.Float(0).tag(sync=True)
+    """The current time of the frame.
+
+    - Type: `float`, optional
+    - Default: `0`
+    """
+
+    get_color = ColorAccessor(None, allow_none=True)
+    """
+    The color of each path in the format of `[r, g, b, [a]]`. Each channel is a number
+    between 0-255 and `a` is 255 if not supplied.
+
+    - Type: [ColorAccessor][lonboard.traits.ColorAccessor], optional
+        - If a single `list` or `tuple` is provided, it is used as the color for all
+          paths.
+        - If a numpy or pyarrow array is provided, each value in the array will be used
+          as the color for the path at the same row index.
+    - Default: `[0, 0, 0, 255]`.
+    """
+
+    get_width = FloatAccessor(None, allow_none=True)
+    """
+    The width of each path, in units specified by `width_units` (default `'meters'`).
+
+    - Type: [FloatAccessor][lonboard.traits.FloatAccessor], optional
+        - If a number is provided, it is used as the width for all paths.
+        - If an array is provided, each value in the array will be used as the width for
+          the path at the same row index.
+    - Default: `1`.
+    """
+
+    get_timestamps = TimestampAccessor(None, allow_none=True)
+    """
+    The timestamp of each coordinate.
+
+    - Type: [TimestampAccessor][lonboard.traits.TimestampAccessor]
+    """
+
+    def __init__(
+        self,
+        *,
+        table: ArrowStreamExportable,
+        get_timestamps: ArrowStreamExportable,
+        _rows_per_chunk: Optional[int] = None,
+        **kwargs: Unpack[TripsLayerKwargs],
+    ):
+        super().__init__(
+            table=table,
+            _rows_per_chunk=_rows_per_chunk,
+            get_timestamps=get_timestamps,  # type: ignore
+            **kwargs,
+        )
+
+    @classmethod
+    def from_movingpandas(
+        cls,
+        traj_collection: TrajectoryCollection,
+        **kwargs: Unpack[TripsLayerKwargs],
+    ) -> Self:
+        import shapely
+
+        num_coords = 0
+        num_rows = len(traj_collection)
+        offsets = np.zeros(num_rows + 1, dtype=np.int32)
+
+        for i, traj in enumerate(traj_collection.trajectories):
+            num_coords += traj.size()
+            offsets[i + 1] = num_coords
+
+        coords = np.zeros((num_coords, 2), dtype=np.float64)
+        timestamps = np.zeros(num_coords, dtype=np.int64)
+
+        for i, traj in enumerate(traj_collection.trajectories):
+            start_offset = offsets[i]
+            end_offset = offsets[i + 1]
+
+            # millisecond-based timestamps
+            int64_ms_timestamps = traj.df.index.to_series().astype(np.int64) // (
```

Comment:
on the project I'm working on the timestamps in the parquet files (one every 3s) are stored at ms resolution so the next line creates duplicates. perhaps:
```suggestion
            int64_ms_timestamps = traj.df.index.as_unit("ns").to_series().astype(np.int64) // (
```
or a warning in the docs?

Reply:
This PR no longer casts for the generic MovingPandas to GeoArrow conversion, and instead keeps whatever the existing time unit MovingPandas had. But we'll still have to do some conversion for Lonboard because the GPU uses float32 so we don't have enough precision to render nanosecond integers

Reply:
@kdpenner you might be interested in looking at [`ca9dcd1` (#292)](https://github.com/developmentseed/lonboard/pull/292/commits/ca9dcd1939ccf0d0cd73efcbd9d2718aa38656f8). Because deck.gl stores timestamps as float32 values, we're forced to ensure that the input data can be represented as integers within the float32 integer range. (Technically we don't need _integers_ within float32 values, but we're losing precision _somewhere_ regardless).

So if you pass in data with nanosecond precision and it has a larger range than can fit in float32 integers (33554432) then a warning will be printed and the input data will automatically be reduced in precision, as far as to seconds.